requirements.txt

```
@@ -1,4 +1,4 @@
-qiskit-terra>=0.22.*
+git+https://github.com/Qiskit/qiskit-terra@main#egg=qiskit-terra
```

```suggestion
qiskit-terra>=0.23.*
```

This is the correct way to update it. This will ensure that the CI works as expected _except_ for the Neko test case.
We had a similar situation [here](https://github.com/Qiskit/qiskit-nature/pull/856#issuecomment-1263707765) where we bumped to `0.22.*` one day before `0.22.0rc1` got released.

I would like to see the CI pass with `0.23.*` specified and have only Neko fail. If that happens, I would be okay with merging this (or we wait until we are closer to `0.23.0rc1` being released, but the PR is then effectively ready)